go/libraries/doltcore/sqle/dtables/conflicts_tables.go

```diff
@@ -93,13 +126,195 @@ func (ct ConflictsTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error)
+		if err != nil {
+			return nil, err
+		}
+		r[itr.b+itr.kd.Count()+i] = f
+	}
+	tup = v.OurValue()
+	for i := 0; i < itr.oursVD.Count(); i++ {
+		f, err := index.GetField(itr.oursVD, i, tup)
+		if err != nil {
+			return nil, err
+		}
+		r[itr.o+itr.kd.Count()+i] = f
+	}
+	tup = v.TheirValue()
+	for i := 0; i < itr.theirsVD.Count(); i++ {
+		f, err := index.GetField(itr.theirsVD, i, tup)
+		if err != nil {
+			return nil, err
+		}
+		r[itr.t+itr.kd.Count()+i] = f
+	}
+
+	return r, nil
+}
+
+func (itr prollyConflictRowIter) Close(ctx *sql.Context) error {
+	return nil
+}
+
+type prollyConflictDeleter struct {
+	kd             val.TupleDesc
+	kB             *val.TupleBuilder
+	pool           pool.BuffPool
+	ed             prolly.ConflictEditor
+	ct             ConflictsTable
+	rs             RootSetter
+	conflictSchema conflict.ConflictSchema
+}
+
+func newProllyConflictDeleter(ct ConflictsTable) *prollyConflictDeleter {
+	conflictMap := durable.ProllyMapFromConflictIndex(ct.confIdx)
+	kd, _, _, _ := conflictMap.Descriptors()
+	ed := conflictMap.Editor()
+	kB := val.NewTupleBuilder(kd)
+	p := conflictMap.Pool()
+	return &prollyConflictDeleter{
+		kd:             kd,
+		kB:             kB,
+		pool:           p,
+		ed:             ed,
+		ct:             ct,
+		conflictSchema: ct.confSch,
+	}
+}
+
+func (cd *prollyConflictDeleter) Delete(ctx *sql.Context, r sql.Row) error {
+	// first columns are the keys
+	for i := 0; i < cd.kd.Count(); i++ {
+		err := index.PutField(cd.kB, i, r[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	key := cd.kB.Build(cd.pool)
+	err := cd.ed.Delete(ctx, key)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StatementBegin implements the interface sql.TableEditor. Currently a no-op.
+func (cd *prollyConflictDeleter) StatementBegin(ctx *sql.Context) {}
+
+// DiscardChanges implements the interface sql.TableEditor. Currently a no-op.
+func (cd *prollyConflictDeleter) DiscardChanges(ctx *sql.Context, errorEncountered error) error {
+	return nil
+}
+
+// StatementComplete implements the interface sql.TableEditor. Currently a no-op.
+func (cd *prollyConflictDeleter) StatementComplete(ctx *sql.Context) error {
+	return nil
+}
+
+// Close finalizes the delete operation, persisting the result.
+func (cd *prollyConflictDeleter) Close(ctx *sql.Context) error {
+	conflicts, err := cd.ed.Flush(ctx)
+	if err != nil {
+		return err
+	}
+
+	// TODO (dhruv): move this code into some kind of ResolveConflicts function
+	var updatedTbl *doltdb.Table
+	if conflicts.Count() == 0 {
+		updatedTbl, err = cd.ct.tbl.ClearConflicts(ctx)
+		if err != nil {
+			return err
+		}
+	} else {
+		updatedTbl, err = cd.ct.tbl.SetConflicts(ctx, cd.conflictSchema, durable.ConflictIndexFromProllyMap(conflicts))
+		if err != nil {
+			return err
+		}
+	}
+
+	updatedRoot, err := cd.ct.root.PutTable(ctx, cd.ct.tblName, updatedTbl)
+	if err != nil {
+		return err
+	}
+
+	return cd.ct.rs.SetRoot(ctx, updatedRoot)
```

Comment:
This doesn't seem right... in particular, you can delete from more than one table in a single statement. We really need root updates to be restricted to write sessions, not individual table editors. For now just add a TODO